wallet/chain/p/client.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p
+
+import (
+	"errors"
+
+	"github.com/ava-labs/avalanchego/vms/platformvm"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/wallet/chain/p/wallet"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+)
+
+var (
+	ErrNotCommitted = errors.New("not committed")
+
+	_ wallet.Client = (*client)(nil)
+)
+
+func NewClient(
+	c platformvm.Client,
+	b wallet.Backend,
+) wallet.Client {
+	return &client{
+		client:  c,
+		backend: b,
+	}
+}
+
+type client struct {
```

Comment:
The wrapping function is nice to enforce that they are provided imo (as the struct is not useful without populating them).

If this signature looks weird I'd be down to unexport the `Client` type entirely to avoid invalid use.